### ISSUE TITLE
[rllib] A3C gradient magnitude is dependent on batch size

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#2506]
### Describe the problem

In https://github.com/ray-project/ray/blob/master/python/ray/rllib/agents/a3c/a3c_tf_policy_graph.py#L28 we use tf.reduce_sum() instead of reduce_mean, which makes the gradient magnitude dependent on batch size.

I also noticed that grad_gnorm is often hitting the clip value of 40.0.

@richardliaw what's the right fix here? Are the lr and grad clip value tuned for batch_size=20, in which case we should divide the by 20 and switch to reduce_mean?
